### ISSUE TITLE
[Bug] Scripts don't accept & operator

### ISSUE THREAD

**Issue by benwainwright** [yarnpkg/berry#1349]
**Describe the bug**

In order to run cypress tests, I use the & operator to background the execution of the process as described [here](https://docs.cypress.io/guides/guides/continuous-integration.html#Boot-your-server). This worked fine in Yarn 1, but in Yarn 2 if any script contains this operator I get

```
Syntax Error: Expected "$", "$(", "${", "&&", "'", ";", "<", "<<<", ">", ">>", "\"", "\\", "|", "|&", "||", [ \t], or end of input but "&" found (line 1, column 10).
    at tt (/Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:28:320521)
    at Object.parse (/Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:28:330627)
    at Object.t.parseShell (/Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:28:315544)
    at Object.t.execute (/Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:16:19878)
    at A.configuration.reduceHook.script (/Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:16:5595)
    at /Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:16:5724
    at async s.mktempPromise (/Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:2:5648)
    at async Object.B [as executePackageScript] (/Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:16:5355)
    at async c.execute (/Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:50:58654)
    at async c.validateAndExecute (/Users/wainwb01/repos/planv7/.yarn/releases/yarn-berry.js:16:50176)   
```
                                                                          
**To Reproduce**
Add "&" to any "script" in your package.json then run the script

**Comment by IT-CASADO** [comment]
Same on my side on Windows: Following script is failing:

**eslint --cache --fix ./ >/dev/null 2>&1 || true**

Error: 

Syntax Error: Expected "$", "$(", "${", "'", "\"", "\\", or [ \t] but "&" found (line 1, column 38).

**Comment by altrugon** [comment]
I'm experiencing the same issue, my environment:

```
node --version && yarn --version
v12.16.3
2.1.1
```

So will be the entire Drupal community if they try to run their "build" or "watch" script from this [package.json](https://git.drupalcode.org/project/drupal/blob/0bada7896a4b4e18fe6b1145f24620bd7dbbbc3d/core/package.json) with Yarn 2.

I hope this get a fix soon. Thank you so much for all the hard work.

**Comment by paul-soporan** [comment]
Note that until we implement support for the `&` operator in our portable shell (which guarantees that things behave the same way on all platforms - e.g. environment variables - `YARN_CHECKSUM_BEHAVIOR=ignore yarn install`), you can run the script with another shell as an escape hatch:
```json
{
  "scripts": {
    "build": "bash -c 'yarn build:css & yarn build:js'"  
  }
}
```

**Comment by joshuapinter** [comment]
Just want to chime in to add that if you have single quotes in your script/command, you can just escape double quotes like this: 

```js
"build": "bash -c \"yarn build:css & yarn build:js\"" 
```

**Comment by TrySpace** [comment]
This is indeed a Windows issue, I've has this exact thing on multiple Windows machines, and on Linux it works fine. I'm not sure how to search for documentation on this, but it must be a common issue people encounter when running `package.json scripts

The `&` is simply not supported by the Windows terminal logic.

Sources:

> Note2: & operator does not work on Windows' cmd.exe
https://github.com/mysticatea/npm-run-all/blob/master/docs/npm-run-all.md
https://stackoverflow.com/questions/39172536/running-npm-scripts-sequentially/39172660#39172660


Some other interesting things to check on:
https://stackoverflow.com/questions/2416662/what-are-the-powershell-equivalents-of-bashs-and-operators
https://superuser.com/questions/345602/what-is-cmds-equivalent-to-bashs-ampersand-for-running-a-command-without-w
https://www.google.nl/search?q=windows+terminal+single+ampersand+concurrency

**Comment by merceyz** [comment]
> This is indeed a Windows issue

Outside of Yarn it is indeed but in this case we ship a custom shell that (now) implements support for `&`

**Comment by TrySpace** [comment]
> > This is indeed a Windows issue
> 
> Outside of Yarn sure, but in this case we ship a custom shell that (now) implements support for `&`

Nice! from which version of Yarn is this supported?

**Comment by merceyz** [comment]
The PR that closed this issue was released in `3.1.0`

**Comment by TrySpace** [comment]
Any idea when this will be merged into the main yarn codebase?

**Comment by paul-soporan** [comment]
It has been merged into the main Yarn codebase 7 months ago and it has been available in Yarn stable since `3.1.0`.
